### PR TITLE
 Add UI for enrolling in Feature Configuration

### LINKED ISSUE #5: Add UI for enrolling in Feature Configuration
We want to be able to copy + paste an Experiment feature configuration JSON into the addon to enroll in an experiment. This requires the following:

- Some UI: a textarea to enter the JSON and a button to enroll when clicked
- Additions to src/apis/nimbus.api and nimbus.json that take a JSON feature configuration, generate an experiment JSON, and enroll in it. (You can crib off NimbusTestUtils' [`ExperimentFakes.enrollWithFeatureConfig`](https://searchfox.org/mozilla-central/rev/7fe1954b761abeff36122b4a6ac74619704ee787/toolkit/components/nimbus/test/NimbusTestUtils.sys.mjs#249) to see how it generates recipes.)

### PR DESCRIPTION
**Pull request by RJAK11**
Implemented feature to allow users to paste an Experiment feature configuration JSON and select a feature ID to enroll into an experiment.

Fixes #5

### CODE REVIEW

File: src/ui/components/FeatureConfigPage.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const handleInputChange = useCallback((event: ChangeEvent<HTMLTextAreaElement>) => {
+    setJsonInput(event.target.value);
+  }, []);
+
+  const handleFeatureSelect = useCallback((featureId) => {
+    setSelectedFeatureId(featureId);
+  }, []);
+
+  const handleEnrollClick = useCallback(async () => {
+    const jsonData = jsonInput;
```
Comment (brennie) [inline src/ui/components/FeatureConfigPage.tsx:17]: Why rebind here?

**Review by brennie — CHANGES_REQUESTED** [review]
There are some formatting issues that will be fixed by rebasing on top of main and running `npm run fix`

File: src/ui/components/FeatureConfigPage.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    try {
```
[the remark this answers is not in the document]
Comment (RJAK11) [reply src/ui/components/FeatureConfigPage.tsx:29]: I think both `JSON.parse(jsonInput)` and `browser.experiments.nimbus.enrollWithFeatureConfig` can throw errors so I assumed that we should use try/catch

File: src/apis/nimbus.js [hunk 3]
```diff
@@ -26,6 +29,58 @@ var nimbus = class extends ExtensionAPI {
               throw error;
             }
           },
+
+          async enrollWithFeatureConfig(featureId, jsonData, isRollout) {
```
Comment (brennie) [inline src/apis/nimbus.js:33]: lets rename `jsonData` to `featureValue`

File: src/types/global.d.ts [hunk 4]
```diff
@@ -3,6 +3,21 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+declare module "*.png" {
+  const value: string;
+  export default value;
+}
```
Comment (brennie) [inline src/types/global.d.ts:11]: neat!

File: src/ui/components/FeatureConfigPage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (brennie) [reply src/ui/components/FeatureConfigPage.tsx:29]: Lets move the try/catch to just around the JSON.parse and API call.

File: src/ui/index.css [hunk 5]
```diff
@@ -3,11 +3,11 @@
 body, html {
     margin: 0;
     padding: 0;
-    font-family: Arial, sans-serif; 
+    font-family: "Segoe UI", Tahoma, Verdana, sans-serif;
```
Comment (brennie) [inline src/ui/index.css:6]: Lets use: `"Helvetica Neue", Helvetica, Arial, sans-serif;`

File: src/ui/index.css [hunk 6]
```diff
@@ -3,11 +3,11 @@
...
 }
 
 .navbar {
-    background-color: #382459;
+    background-color: #44144c;
```
Comment (brennie) [inline src/ui/index.css:10]: In a followup we should clean up the CSS to use constants for a lot of these, but we're in the prototype phase so lets not worry now.

**Review by brennie — APPROVED** [review]
LGTM

**Pull request merged**